File: climbing-stairs/climbing-stairs.cpp
class Solution {
public:
    int climbStairs(int n) {
        if(n<=2) return n;
        int dp[n+1];
        dp[0]=0;
        dp[1]=1;
        dp[2]=2;
        int i=3;
        while(i<=n)
        {
            dp[i]=dp[i-1]+dp[i-2];
            i++;
        }
        return dp[n];
    }
};

File: 70-climbing-stairs/climbing-stairs.java
class Solution {
    public int climbStairs(int n) {
        int prev1 = 1;
        int prev2 = 1;
        int currWays = 1;
        for(int i=2;i<=n;i++)
        {
            currWays = prev1+prev2;
            prev1 = prev2;
            prev2 = currWays;
        }
        return currWays;
    }
}